src/Azure.Functions.Cli/Actions/AzureActions/LoginAction.cs

```diff
@@ -1,23 +1,63 @@
 ﻿using System;
-using System.Linq;
+using System.Text;
 using System.Threading.Tasks;
-using Colors.Net;
 using Azure.Functions.Cli.Arm;
 using Azure.Functions.Cli.Interfaces;
+using Fclp;
 
 namespace Azure.Functions.Cli.Actions.AzureActions
 {
     [Action(Name = "login", Context = Context.Azure, HelpText = "Log in to an Azure account")]
     class LoginAction : BaseAzureAccountAction
     {
+        private string _username = string.Empty;
+        private string _password = string.Empty;
+
         public LoginAction(IArmManager armManager, ISettings settings)
             : base(armManager, settings)
         { }
 
+        public override ICommandLineParserResult ParseArgs(string[] args)
+        {
+            Parser.Setup<string>('u')
+                .WithDescription("username")
+                .Callback(username => _username = username);
+
+            Parser.Setup<string>('w')
+                .WithDescription("password")
+                .Callback(password => _password = password);
+            
+            return base.ParseArgs(args);
+        }
+
         public override async Task RunAsync()
         {
-            await _armManager.LoginAsync();
+            if (string.IsNullOrWhiteSpace(_username))
+                await _armManager.LoginAsync();
+            else
+            {
+                if (string.IsNullOrWhiteSpace(_password))
+                    _password = PromptForPassword();
+
+                await _armManager.LoginAsync(_username, _password);
+            }
+
             await PrintAccountsAsync();
         }
+
+        private static string PromptForPassword()
+        {
+            Console.Write("password: ");
+            var password = new StringBuilder();
+            while (true)
+            {
+                var key = Console.ReadKey(true);
+                if (key.Key == ConsoleKey.Enter)
+                    break;
+                password.Append(key.KeyChar);
+            }
+            Console.Write(Environment.NewLine);
+            return password.ToString();
```

Comment:
There is [`SecurityHelpers.ReadPassword()`](https://github.com/Azure/azure-functions-cli/blob/master/src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs#L73) which is similar to this method.